index.d.ts

```diff
@@ -1,10 +1,15 @@
 declare function deepmerge<T>(x: Partial<T>, y: Partial<T>, options?: deepmerge.Options): T;
 declare function deepmerge<T1, T2>(x: Partial<T1>, y: Partial<T2>, options?: deepmerge.Options): T1 & T2;
 
+declare interface CustomMerge<T1, T2> {
+  (x: Partial<T1>, y: Partial<T1 | T2>): T1 | T1 & T2;
+}
+
 declare namespace deepmerge {
 	export interface Options {
 		arrayMerge?(target: any[], source: any[], options?: Options): any[];
 		clone?: boolean;
+		customMerge?: (key: string, options?: Options) => (x: any, y: any) => any | undefined;
```

Comment:
I suppose this is valid, but that `|` at the end is messing with my internal order-of-operations processor...  are these parentheses valid I wonder?
```suggestion
		customMerge?: (key: string, options?: Options) => ((x: any, y: any) => any) | undefined;
```

Reply:
I think that this makes more sense, I'll run it through the tests and if it's valid, I'll commit it up

Reply:
Added a commit to cover this